### DISCUSSION TITLE
Hybrid attribute expression querying one to many relationship

### DISCUSSION THREAD

**Discussion by sponsfreixes** [sqlalchemy/sqlalchemy#6448] [Usage Questions] [answered]
I'm having difficulties building the expression for an hybrid attribute that checks the field on one element of a related model. I think the sample code will show better what I am trying to accomplish:

```python
class Foo(Model):
    __tablename__ = "foo"

    set_action = "set"
    unset_action = "unset"

    id = Column(Integer, primary_key=True)
    created = Column(DateTime, nullable=False, server_default=func.now())
    bar_id = Column(Integer, ForeignKey("bar.id"), nullable=False)
    action = Column(String(DEFAULT_STRING_LENGTH), nullable=False)


class Bar(Model):
    __tablename__ = "bar"

    id = Column(Integer, primary_key=True)
    name = Column(String(DEFAULT_STRING_LENGTH), nullable=False)

    foos = relationship(
        "Foo",
        lazy="selectin",
        order_by="Foo.created",
        backref=backref("bar", lazy=True),
    )

    @hybrid_property
    def latest_foo_is_set(self):
        if not self.foos:
            return False
        return self.foos[-1].action == Foo.set_action

    # THIS IS THE EXPRESSION NOT WORKING AS INTENDED
    @latest_foo_is_set.expression
    def latest_foo_is_set(cls):
        return session.execute(
            select(Foo.action)
            .where(cls.id == Foo.bar_id)
            .order_by(Foo.created.desc())
            .limit(1)
        ).scalar() == Foo.set_action

```

The code above does not break if you query something like ` db.session.query(Bar).filter(Bar.latest_foo_is_set).all()`, but I'm not getting the right results either. I get everything back with `filter(Bar.latest_foo_is_set)` and nothing with `filter(~Bar.latest_foo_is_set)`.

I've checked the documentation and other GitHub/Stackoverflow discussions, but I am not able to find an example close enough to what I am trying to achieve. Any tips about what I am doing wrong?

**Comment by CaselIT** [comment, marked as answer]
Hi,

I have little experience with `hybrid_property`, but I believe you need to return a select, not execute a query with it. So your example would become something like 
```
return select(Foo.action).where(cls.id == Foo.bar_id).order_by(Foo.created.desc()).limit(1).scalar_subquery() == Foo.set_action
```

**Reply by sponsfreixes** [reply]
Thanks, that did it! In order to account for the case where there were no Foo, I added a coalesce so that it would return `False` instead of `None`:

```python
return func.coalesce(select(Foo.action).where(cls.id == Foo.bar_id).order_by(Foo.created.desc()).limit(1).scalar_subquery() == Foo.set_action, False)
```
